File: sync-chat-client-go/src/ui/chatLog.go
package ui

import (
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

// chatLog es la caja blanca de mensajes de la imagen: líneas que se
// apilan hacia abajo y hacen autoscroll. La usan tanto la ventana de
// grupo como cada ventana privada, así que vive aparte para no repetir
// este layout en las dos.
type chatLog struct {
	box  *fyne.Container
	view *container.Scroll
}

func newChatLog() *chatLog {
	box := container.NewVBox()
	return &chatLog{box: box, view: container.NewVScroll(box)}
}

func (l *chatLog) append(line string) {
	l.box.Add(widget.NewLabel(line))
	l.box.Refresh()
	l.view.ScrollToBottom()
}

File: sync-chat-client-py/src/ui/chat_log.py
"""ChatLog: la caja blanca de mensajes de la imagen. Líneas que se apilan
hacia abajo y hacen autoscroll. La comparten la ventana de grupo y cada
privada, igual que chatLog en el cliente de Go."""

from __future__ import annotations

import tkinter as tk
from tkinter import scrolledtext


class ChatLog(scrolledtext.ScrolledText):
    def __init__(self, master: tk.Widget):
        super().__init__(master, state="disabled", wrap="word")

    def append(self, line: str) -> None:
        self.configure(state="normal")
        self.insert("end", line + "\n")
        self.configure(state="disabled")
        self.see("end")
